app/src/org/commcare/android/fragments/SetupEnterURLFragment.java

```diff
@@ -133,14 +133,17 @@ public void onResume() {
     }
 
     @Override
-    public void onAttach(Activity activity) {
-        activity.getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_ALWAYS_VISIBLE);
+    public void onAttach(Context context) {
+        super.onAttach(context);
 
-        super.onAttach(activity);
-        if (!(activity instanceof URLInstaller)) {
-            throw new ClassCastException(activity + " must implemement " + interfaceName);
+        if (context instanceof Activity) {
+            ((Activity)context).getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_ALWAYS_VISIBLE);
+        }
```

Comment:
Sorta feels strange having an `instanceof Activity` check. Might be possible to set this input mode somewhere else, or by other means, in a more natural way?